smithy-model/src/main/java/software/amazon/smithy/model/transform/ChangeShapeType.java

```diff
@@ -96,6 +113,17 @@ public Shape integerShape(IntegerShape shape) {
             return copyToSimpleShape(to, shape);
         }
 
+        @Override
+        public Shape intEnumShape(IntEnumShape shape) {
+            if (to.getCategory() != ShapeType.Category.SIMPLE) {
+                throw invalidType(shape, to, "Enum types can only be converted to simple types.");
```

Comment:
No, this can be converted to other simple types too. The string enum checks for it explicitly because it needs to do some extra leg work when the target is a string, but that isn't the case for intEnum since it doesn't need to mess with the enum trait.